### ISSUE TITLE
Window.top calls fail if site is a child in an iframe

### ISSUE THREAD

**Issue by animepauly** [fritz-c/react-image-lightbox#102]
I am using this component in a site that I am embedding into another site. When the child site stands alone there is no issue but when it's embedded I get this error:
```
Uncaught DOMException: Blocked a frame with origin "my-child-site.com" from accessing a cross-origin frame.
```
Could you do a check if the component is in a child site to not use window.top and maybe window.self or another solution instead?

I can't share my code as it's not released but if need be I can make a whole new example to show you. Thanks!

**Comment by animepauly** [comment]
I can confirm that window.self works if the site is a child of an iframe. Can you run a check if the site is a child and to use self instead of top? That should fix the issue.

**Comment by fritz-c** [comment]
Fixed as of v4.6.0.
